applications/mne_scan/plugins/ftbuffer/ftconnector.h

```diff
@@ -0,0 +1,373 @@
+     */
+    bool parseHeaderDef(QBuffer &readBuffer);
+
+    //=========================================================================================================
+    /**
+     * @brief Parses messadef and returns bufsize
+     *
+     * @param readBuffer - QBuffer with return messagedef_t data from buffer
+     *
+     * @return returns messagedef_t.bufsize
+     */
+    int parseMessageDef(QBuffer &readBuffer);
+
+    //=========================================================================================================
+    /**
+     * @brief Parses datadef and returns bufsize
+     *
+     * @param dataBuffer - QBuffer with return datadef_t data from buffer
+     *
+     * @return returns datadef_t.bufsize
+     */
+    int parseDataDef(QBuffer &dataBuffer);
+
+    //=========================================================================================================
+    /**
+     * @brief Parses sample data received from buffer, formates it and saves it to m_pMatEmit;
+     *
+     * @param datasampBuffer - QBuffer with return data from buffer
+     * @param bufsize - bufsize of sample data
+     *
+     * @return true if successful, false if unsuccessful
+     */
+    bool parseData(QBuffer &datasampBuffer,
+                   int bufsize);
+
+    //=========================================================================================================
+    /**
+     * @brief Opens Buffer, reads numBytes from socket and sets index to zero
+     *
+     * @param buffer - QBuffer to which daa will be written
+     * @param numBytes - how many bytes to read from socket
+     */
+    void prepBuffer(QBuffer &buffer,
+                    int numBytes);
+
+    //=========================================================================================================
+    /**
+     * @brief Returns total amount of samples written to buffer
+     *
+     * @return returns total amount of samples written to buffer
+     */
+    int totalBuffSamples();
+
+    int                                     m_iNumSamples;                          /**< Number of samples we've read from the buffer */
+    int                                     m_iNumNewSamples;                       /**< Number of total samples (read and unread) in the buffer */
+    int                                     m_iMsgSamples;                          /**< Number of samples in the latest buffer transmission receied */
+    int                                     m_iNumChannels;                         /**< Number of channels in the buffer data */
+    int                                     m_iDataType;                            /**< Type of data in the buffer */
+    int                                     m_iNeuromagHeader;                      /**< Size of neuromag header chunk */
+    int                                     m_iPort             = 1972;             /**< Port where the ft bufferis found */
```

Comment:
Please perform init in constructor only, see [here](https://stackoverflow.com/questions/28413154/c-initializing-variables-in-header-vs-with-constructor/28414747). We decided on using the "init in constructor" method.